Install Failure : boost::variant does not contain specified type U

On ubuntu Xenial

```
Z3 SMT solver found. This enables optional SMT checking with Z3.
-- Configuring done
-- Generating done
-- Build files have been written to: /root/solidity/build
[  4%] Built target jsoncpp-project
[  4%] Built target solidity_BuildInfo.h
[  9%] Built target devcore
[ 18%] Built target evmasm
[ 21%] Built target lll
[ 22%] Built target lllc
[ 23%] Building CXX object libsolidity/CMakeFiles/solidity.dir/__/libjulia/optimiser/DataFlowAnalyzer.cpp.o
[ 23%] Building CXX object libsolidity/CMakeFiles/solidity.dir/__/libjulia/optimiser/FullInliner.cpp.o
In file included from /usr/include/boost/type_index/stl_type_index.hpp:32:0,
                 from /usr/include/boost/type_index.hpp:29,
                 from /usr/include/boost/variant/variant.hpp:21,
                 from /usr/include/boost/variant.hpp:17,
                 from /root/solidity/libsolidity/inlineasm/AsmDataForward.h:25,
                 from /root/solidity/libjulia/ASTDataForward.h:24,
                 from /root/solidity/libjulia/optimiser/FullInliner.h:22,
                 from /root/solidity/libjulia/optimiser/FullInliner.cpp:21:
/usr/include/boost/variant/get.hpp: In instantiation of ‘typename boost::add_reference<T>::type boost::strict_get(boost::variant<T0, TN ...>&) [with U = dev::solidity::assembly::ExpressionStatement&; T0 = dev::solidity::assembly::ExpressionStatement; TN = {dev::solidity::assembly::Instruction, dev::solidity::assembly::Label, dev::solidity::assembly::StackAssignment, dev::solidity::assembly::Assignment, dev::solidity::assembly::VariableDeclaration, dev::solidity::assembly::FunctionDefinition, dev::solidity::assembly::If, dev::solidity::assembly::Switch, dev::solidity::assembly::ForLoop, dev::solidity::assembly::Block}; typename boost::add_reference<T>::type = dev::solidity::assembly::ExpressionStatement&]’:
/usr/include/boost/variant/get.hpp:284:25:   required from ‘typename boost::add_reference<T>::type boost::get(boost::variant<T0, TN ...>&) [with U = dev::solidity::assembly::ExpressionStatement&; T0 = dev::solidity::assembly::ExpressionStatement; TN = {dev::solidity::assembly::Instruction, dev::solidity::assembly::Label, dev::solidity::assembly::StackAssignment, dev::solidity::assembly::Assignment, dev::solidity::assembly::VariableDeclaration, dev::solidity::assembly::FunctionDefinition, dev::solidity::assembly::If, dev::solidity::assembly::Switch, dev::solidity::assembly::ForLoop, dev::solidity::assembly::Block}; typename boost::add_reference<T>::type = dev::solidity::assembly::ExpressionStatement&]’
/root/solidity/libjulia/optimiser/FullInliner.cpp:171:75:   required from here
/usr/include/boost/variant/get.hpp:212:5: error: static assertion failed: boost::variant does not contain specified type U, call to boost::get<U>(boost::variant<T...>&) will always throw boost::bad_get exception
     BOOST_STATIC_ASSERT_MSG(
     ^
/usr/include/boost/variant/get.hpp: In instantiation of ‘typename boost::add_reference<T>::type boost::strict_get(boost::variant<T0, TN ...>&) [with U = dev::solidity::assembly::FunctionalInstruction&; T0 = dev::solidity::assembly::FunctionalInstruction; TN = {dev::solidity::assembly::FunctionCall, dev::solidity::assembly::Identifier, dev::solidity::assembly::Literal}; typename boost::add_reference<T>::type = dev::solidity::assembly::FunctionalInstruction&]’:
/usr/include/boost/variant/get.hpp:284:25:   required from ‘typename boost::add_reference<T>::type boost::get(boost::variant<T0, TN ...>&) [with U = dev::solidity::assembly::FunctionalInstruction&; T0 = dev::solidity::assembly::FunctionalInstruction; TN = {dev::solidity::assembly::FunctionCall, dev::solidity::assembly::Identifier, dev::solidity::assembly::Literal}; typename boost::add_reference<T>::type = dev::solidity::assembly::FunctionalInstruction&]’
/root/solidity/libjulia/optimiser/FullInliner.cpp:174:89:   required from here
/usr/include/boost/variant/get.hpp:212:5: error: static assertion failed: boost::variant does not contain specified type U, call to boost::get<U>(boost::variant<T...>&) will always throw boost::bad_get exception
libsolidity/CMakeFiles/solidity.dir/build.make:1862: recipe for target 'libsolidity/CMakeFiles/solidity.dir/__/libjulia/optimiser/FullInliner.cpp.o' failed
make[2]: *** [libsolidity/CMakeFiles/solidity.dir/__/libjulia/optimiser/FullInliner.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:273: recipe for target 'libsolidity/CMakeFiles/solidity.dir/all' failed
make[1]: *** [libsolidity/CMakeFiles/solidity.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
Failed to build
```

--- a/libjulia/optimiser/FullInliner.cpp
+++ b/libjulia/optimiser/FullInliner.cpp
@@ -168,10 +168,10 @@ void InlineModifier::visit(Statement& _statement)
 	// Replace pop(0) expression statemets (and others) by empty blocks.
 	if (_statement.type() == typeid(ExpressionStatement))
 	{
-		ExpressionStatement& expSt = boost::get<ExpressionStatement&>(_statement);
+		ExpressionStatement& expSt = boost::get<ExpressionStatement>(_statement);
 		if (expSt.expression.type() == typeid(FunctionalInstruction))
 		{
-			FunctionalInstruction& funInstr = boost::get<FunctionalInstruction&>(expSt.expression);
+			FunctionalInstruction& funInstr = boost::get<FunctionalInstruction>(expSt.expression);
 			if (funInstr.instruction == solidity::Instruction::POP)
 				if (MovableChecker(funInstr.arguments.at(0)).movable())
 					_statement = Block{expSt.location, {}};

